### ISSUE TITLE
Choosing a preset is very frustrating

### ISSUE THREAD

**Issue by diovudau** [iurie-sw/geonkick#62]
Running Geonkick for the first time.
Presets are the first (and some times only) usage of a synth.

Choosing is a preset could be much more streamlined here.
The software compiled itself. Both in standalone and in plugin mode it knows where its files are. Presets should be a drop-down list easily accesible. 

But the process right now was not pleasent in itself. It is a multi-step, multi-minute endeavour that should only take two clicks and a second.

First I had to look up in the Readme and Makefile(!) where the presets are on my disk. Actually, I had to run `find /usr | grep gkick` and found out  /usr/share/doc/geonkick/presets  . Why documentation? well then..

Using the file manager to navigate there was already difficult enough because it is missing normal features like typing a letter to jump to that place in the alphabet or entering a path directly.

Then I saw single instrument and "Time Kit-1.0" I selected it, clicked "open" and the file dialog aborted. Ah, it was not a group-preset but just another directory.  So I pressed "open" in the GUI again and I was back in the home dir! 

Finally, after navigating again, I saw that the Time Kit was just several single instruments as well. At least then it remembered the last directory.

Now I only need to find out how to load all the instruments and map them to keys. I beliebe at least that is covered in the documentation. 

Furthermore, I saw only on github that there is an AVLKit.tar.gz with the readme to uncompress it if you want to use it. Nobody will do that. This should be an automated step in make install.

**Comment by iurienistor** [comment]
> "Then I saw single instrument and "Time Kit-1.0" I selected it, clicked "open" and the file dialog aborted."

Hi, thank you for reporting the problem the issue will be fixed in the next versions.

"Presets should be a drop-down list easily accesible.
But the process right now was not pleasent in itself. It is a multi-step, multi-minute endeavour that should only take two clicks and a second."

To develop presets it takes time and most important experience, it is beyond the scope of this project to provide presets. It was a mistake of mine to rename the folder "examples" into "presets" since they are examples and not presets, the package maintainers should not install them anywhere. 

> First I had to look up in the Readme and Makefile(!) where the presets are on my disk. Actually, I had to run find /usr | grep gkick and found out /usr/share/doc/geonkick/presets . Why documentation? well then..

Geonkick build system as far as I  know is not installing these files anywhere. Probably you have used a pakcage from the repository that installs there examples into doc directory. Actually maybe they are right doc is the right place for examples, I need to move them there.

There was a plan to develop a preset browser, but since almost all hosts provide their own preset browser (plugin state browser), I considered that is not needed for now, maybe in the future.

Geonkick is using a self developed Redkite GUI toolkit because of this reason http://lists.lv2plug.in/pipermail/devel-lv2plug.in/2016-March/001593.html
Due to this, to provide a standard file browser dialog for plugins is a long standing issue that have many technical implications. Most plugins avoid to do this. If you are interested you can read it here about other limitations:
https://github.com/Yoshimi/yoshimi/issues/68#issuecomment-569241756

**Comment by dvzrv** [comment]
Hi! I'm packaging geonkick for Arch Linux.

I'm now installing the preset files to `/usr/share/geonkick/presets`. I would really appreciate if the different presets would remain in the same location in the source tarball for the future (be it `doc/examples` or wherever) :)

**Comment by mxmilkb** [comment]
@iurienistor regardless of why the existing examples exist, I think the premise "Presets are the first (and some times only) usage of a synth." is not wrong. It's a situation where 'presets' and 'examples' become synonymous.

Host behaviour relating to plugin presets varies quite a bit. If there's anything that could be done at the plugin to reduce friction because if that, it would be welcome.

But the standalone version doesn't have a host to possibly provide a preset browser.

There is also the opportunity that further example presets could be submitted.

**Comment by iurienistor** [comment]
@dvzrv I'll put them in doc/examples. But Geonkick don't have factory presets yet, thus, probably /usr/share/doc/geonkick/examples would be more apropriate for now, but I don't know. In order to make a factory preset list, I'll need to ask Geonkick users to share with preset if they want, so, I'll be able to create a list.

**Comment by diovudau** [comment]
As you are surely aware Unfa is using geonkick a lot. Maybe he would provide a kit or two.

**Comment by iurienistor** [comment]
@mixmikb, "regardless of why the existing examples exist, I think the premise "Presets are the first (and some times only) usage of a synth." is not wrong."

Yes, I understand this. I have reopened the issue https://github.com/iurie-sw/geonkick/issues/4 of preset browser, but this will be started to be developed only after v2.5 because until than I want to add more functionality for Redkite, and improve things in Geonkick UI usage, and also prepare Redkite GUI for another project I have in mind. Until than I'll ask Geonkick users to share with presets. 

One problem about including presets in the Geonkick repository is that some presets may be composed of samples like AVL-Drumkit.gkit and they take 16MB if uncompressed, but only 800K compressed. Maybe a solution is to add functionality to build system to uncompressed presets during build and install them.

**Comment by iurienistor** [comment]
I have added also this issue related to factory presets - https://github.com/iurie-sw/geonkick/issues/63

**Comment by iurienistor** [comment]
> As you are surely aware Unfa is using geonkick a lot. Maybe he would provide a kit or two.

Yes, I'll ask @unfa about this, and maybe others on forums, and add this also in README.
